trunner/tools/gdb.py

```diff
@@ -222,3 +222,71 @@ def run(self):
             yield
         finally:
             self._close()
+
+
+class PyocdError(ProcessError):
+    name = "PYOCD"
+
+
+class PyocdProcess:
+    """Handler for Pyocd process"""
+
+    def __init__(
+        self,
+        target,
+        extra_args: Optional[List[str]] = None,
+        load_file: Optional[Union[Path, str]] = None,
+        load_offset: int = 0,
+        host_log: Optional[TextIO] = None,
+        cwd: Optional[Path] = None,
+    ):
+        self.target = target
+        self.extra_args = extra_args
+        self.load_file = load_file
+        self.load_offset = load_offset
+        self.host_log = host_log if host_log is not None else io.StringIO()
+        self.cwd = cwd
+        self.proc = None
+        self.output = ""
+
+    def _spawn(self, args, end_message: str):
+        try:
+            if self.load_offset != 0:
+                args.extend([f"{self.load_file}@{hex(self.load_offset)}"])
+            elif self.load_file:
+                args.extend([f"{self.load_file}"])
```

Comment:
It should be done in load method as it has to be specified only when loading sth

```suggestion
```